src/Oro/Bundle/WorkflowBundle/Controller/Api/Rest/WorkflowController.php

```diff
@@ -151,7 +153,26 @@ protected function getEntityReference($entityClass, $entityId)
     public function transitAction(WorkflowItem $workflowItem, $transitionName)
     {
         try {
-            $this->get('oro_workflow.manager')->transit($workflowItem, $transitionName);
+            /** @var WorkflowManager $workflowManager */
+            $workflowManager = $this->get('oro_workflow.manager');
+
+            /**
+             * \Oro\Bundle\WorkflowBundle\Model\WorkflowManager::transit may produce a DB exception
+             * and bring transaction to the rollback-only state.
+             * App exits after that in production mode, so it's normal behaviour. But during functional tests
+             * this breaks all tests in queue after this one.
+             * So we should check transition availability to exclude exceptional situation.
+             */
+            if (!$workflowManager->isTransitionAvailable($workflowItem, $transitionName)) {
+                throw InvalidTransitionException::stepHasNoAllowedTransition(
+                    $workflowItem->getWorkflowName(),
+                    $workflowItem->getCurrentStep()->getName(),
+                    $transitionName
+                );
+            }
+
+            $workflowManager->transit($workflowItem, $transitionName);
+
```

Comment:
Please, don't put additional logic in controllers, they must be as thin as possible. In this particular case, I think it will be better to skip test case which breaks other tests and don't refactor this method(or just remove workflowManager retrieving from try block, because best practice is one try block must exist for one basic operation). It's a known issue and we plan to refactor such cases in future.